boot/boot.go

```diff
@@ -403,19 +416,44 @@ func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet,
 	return nil
 }
 
-// MakeBootable sets up the image filesystem with the given rootdir
-// such that it can be booted.
-func MakeBootable(model *asserts.Model, rootdir string, bootWith *BootableSet) error {
+func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *BootableSet) error {
+	// XXX: move to dirs ?
+	runMnt := filepath.Join(rootdir, "/run/mnt/")
+
+	// TODO:UC20:
+	// - create grub.cfg instead of using the gadget one
+	// - extract kernel
+	// - write modeenv
+
+	// update recovery grubs grubenv to indicate that we transition
+	// to run mode now
 	opts := &bootloader.Options{
-		// XXX: allow to override this
-		PrepareImageTime: true,
+		// setup the recovery part of the bootloader
+		Recovery: true,
+	}
+	bl, err := bootloader.Find(filepath.Join(runMnt, "ubuntu-seed"), opts)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot find bootloader: %v", err)
+	}
+	blVars := map[string]string{
+		"snapd_recovery_mode": "run",
+	}
+	if err := bl.SetBootVars(blVars); err != nil {
+		return fmt.Errorf("cannot set recovery system environment: %v", err)
 	}
 
+	return nil
+}
+
+// MakeBootable sets up the image filesystem with the given rootdir
```

Comment:
// MakeBootable sets up the given bootable set and target filesystem such that the system can be booted.
// rootdir points to an image filesystem (UC 16/18), image recovery filesystem (UC20 at prepare-image time) or ephemeral system (UC20 install mode).